packages/flutter/test/material/carousel_test.dart

```diff
@@ -1696,6 +1696,48 @@ void main() {
     expect(secondItemWidth, surfaceWidth * secondWeight / totalWeight);
   });
 
+  testWidgets('CarouselView.builder creates items lazily', (WidgetTester tester) async {
+    final List<int> builtItems = <int>[];
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: CarouselView.builder(
+            itemExtent: 300.0,
+            itemCount: 1000,
+            itemBuilder: (BuildContext context, int index) {
+              builtItems.add(index);
+              return Container(
+                color: Colors.blue[index % 9 * 100],
+                child: Center(child: Text('Item $index')),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    // Only visible items should be built initially.
+    expect(builtItems.length, lessThan(10));
+    expect(builtItems, contains(0));
+    expect(builtItems, contains(1));
+
+    // Scroll to a far item.
+    await tester.drag(find.byType(CarouselView), const Offset(-2000.0, 0.0));
+    await tester.pumpAndSettle();
+
+    // Clear built items to see what's built after scrolling.
+    builtItems.clear();
+
+    // Force rebuild by scrolling a bit more.
+    await tester.drag(find.byType(CarouselView), const Offset(-300.0, 0.0));
+    await tester.pump();
+
+    // Should have built new items, not the initial ones.
+    expect(builtItems.isNotEmpty, isTrue);
```

Comment:
I think you can do something like this and get a better error message when a failure happens:

```dart
expect(builtItems, isNotEmpty);
```